File: Python Programs/Exp_13.py
# Python program to perform inorder tree traversal

# Perform inorder traversal of a tree with root 25 , left child 15 and write child 30

class Node:
	def __init__(self, key):
		self.left = None
		self.right = None
		self.val = key
def printInorder(root):
	if root:
		printInorder(root.left)
		print(root.val,end=''),
		printInorder(root.right)
T1=input()
root = Node("25,")
root.left = Node("15,")
root.right = Node("30")
printInorder(root)

# Input :
#  root 25 , left child 15 ,write child 30

# Output :
# 15,25,30

File: CPP Programs/Exp_13.cpp
// C++ program to perform inorder tree traversal

#include <iostream>
using namespace std;

struct Node {
  int data;
  char ch;
  struct Node *left, *right;
  Node(int data, char c) {
    this->data = data;
    this->ch = c;
    left = right = NULL;
  }
};

// Inorder traversal
void inorderTraversal(struct Node* node) {
  if (node == NULL)
    return;

  inorderTraversal(node->left);
  if(node->ch != '\0')
  cout << node->data <<node->ch;
  else
  cout<<node->data;
  inorderTraversal(node->right);
}

int main() {
  struct Node* root = new Node(25, ',');
  root->left = new Node(15, ',');
  root->right = new Node(30, '\0');

  inorderTraversal(root);
}